cmd/root.go

```diff
@@ -0,0 +1,60 @@
+package cmd
+
+import (
+	"fmt"
+
+	"github.com/10gen/realm-cli/internal/cli"
+	clierr "github.com/10gen/realm-cli/internal/cli/err"
+	"github.com/10gen/realm-cli/internal/config"
+	"github.com/10gen/realm-cli/internal/flags"
+	"github.com/10gen/realm-cli/internal/realm"
+	"github.com/10gen/realm-cli/internal/ui"
+	"github.com/10gen/realm-cli/internal/usage"
+
+	"github.com/spf13/cobra"
+	"honnef.co/go/tools/version"
+)
+
+var (
+	rootCmd = &cobra.Command{
+		Version: version.Version,
+		Use:     config.Name,
+		Short:   "CLI tool to manage your MongoDB Realm application",
+		Long:    fmt.Sprintf("Use %s command help for information on a specific command", config.Name),
+	}
+)
+
+// Execute runs the CLI
+func Execute() {
+	clierr.HandleErr(rootCmd.Execute())
+}
+
+func init() {
+	baseCmd, baseCmdErr := newBaseCommand()
+	clierr.HandleErr(baseCmdErr)
+
+	rootCmd.PersistentFlags().StringVarP(&baseCmd.Config.Name, flags.Profile, flags.ProfileShort, config.DefaultName, usage.Profile)
+	rootCmd.PersistentFlags().BoolVar(&baseCmd.UI.DisableColors, flags.DisableColors, false, usage.DisableColors)
+	rootCmd.PersistentFlags().StringVar(&baseCmd.UI.OutputFormat, flags.OutputFormat, ui.DefaultOutputFormat, usage.OutputFormat)
+	rootCmd.PersistentFlags().StringVar(&baseCmd.UI.OutputTarget, flags.OutputTarget, "", usage.OutputTarget)
+	rootCmd.PersistentFlags().StringVar(&baseCmd.RealmConfig.BaseURL, flags.RealmBaseURL, realm.DefaultBaseURL, usage.RealmBaseURL)
+
+	rootCmd.AddCommand(cli.LoginCommand(baseCmd))
+	rootCmd.AddCommand(cli.LogoutCommand(baseCmd))
+	rootCmd.AddCommand(cli.WhoamiCommand(baseCmd))
```

Comment:
in my head, I had these flipped where baseCmd would wrap the underneath commands. That way, the underlying commands wouldn't need to reference the baseCmd at all as it would be a "middleware" of sorts.

Reply:
we now have the `CommandFactory` building these, which inverts the control of a "base command".  I think we're good here now